caffe2/CMakeLists.txt

```diff
@@ -1090,6 +1090,28 @@ if(USE_XPU)
   torch_compile_options(torch_xpu)  # see cmake/public/utils.cmake
   target_compile_options_if_supported(torch_xpu "-Wno-deprecated-copy")  # see cmake/public/utils.cmake
   target_compile_definitions(torch_xpu PRIVATE USE_XPU)
+
```

Comment:
```suggestion
  if(NOT EXISTS "${TORCH_ROOT}/third_party/torch-xpu-ops/.git")
      execute_process(
          COMMAND git clone https://github.com/intel/torch-xpu-ops "${TORCH_ROOT}/third_party/torch-xpu-ops"
      )
  else()
      execute_process(
          COMMAND git -C "${third_party}/torch-xpu-ops" pull
      )
  endif()
```

Reply:
In general, @malfet , are you okay with the suggested change?

Reply:
@malfet, we have applied the change first to follow the soft-check mechanism. We would appreciate it if you had any implementation proposals :)

Reply:
Yes, this sounds fine to me

Reply:
Thanks, we will update the PR and keep you posted.

Reply:
We are working on the code and data separation. However, we have yet to find a proper place to put the commit pin file.  It kind of conflicts with the current PyTorch CMake design if we put the version under the CMake folder directly.

There are two places in the stock PyTorch to host commit pin files for different packages: `.ci` and `.github`. `.ci` is dedicated to ci. Therefore, we add a xpu commit pin to the `.github` folder now.

And any suggestions are appreciated.

Reply:
You can put it in `third_party` folder, which feels like a natural place to keep both data and metadata

Reply:
Good idea! Thanks for the suggestion. We will put the `xpu.txt` under `third_party`

Reply:
@malfet , we have refined the PR. May I know if you have any other comments?